src/components/address/address.tsx

```diff
@@ -0,0 +1,31 @@
+import { useResolveEnsName } from 'eth-hooks/dapps';
+import { TEthersProvider } from 'eth-hooks/models';
+
+export interface AddressProps {
+  address: string;
+  ensProvider?: TEthersProvider;
+  blockExplorer?: string;
+}
+
+export interface AddressResult {
+  shortAddress: string;
+  ensName?: string;
+  explorerLink: string;
+}
+
+const blockExplorerLink = (address: string, blockExplorer?: string): string =>
+  `${blockExplorer || 'https://etherscan.io/'}address/${address}`;
+
+export const useAddress = (props: AddressProps): AddressResult => {
```

Comment:
I would add tsdocs comments similar to eth-hooks.  It helps create automated documentation later.  https://scaffold-eth.github.io/eth-hooks/docs/api/modules/Hooks. @stevenpslade